bindings/java/wasmedge-java/src/main/java/org/wasmedge/NativeResource.java

```diff
@@ -9,14 +9,13 @@
 public abstract class NativeResource implements Closeable {
     private long pointer;
 
-    public NativeResource(long pointer) {
-        this.pointer = pointer;
+    protected NativeResource() {
+        pointer = 0;
     }
```

Comment:
I know that you want to prevent constructing `NativeResource` out of the package, but do we need uninitialized pointer? Or where do we invoke `NativeResource()`?

Reply:
No, we don't need an uninitialized pointer. We just need a default constructor for some classes like `CompilerContext`, which requires calling the super constructor first. 

```
    public CompilerContext(ConfigureContext configureContext) {
        super();
        nativeInit(configureContext);
    }
```

Reply:
Get it, thanks